docs/builtins/General/chain-data.md

```diff
@@ -1,41 +1,78 @@
 ## chain-data
-Use `chain-data` to retrieve transaction public metadata, returning an object with fields including 'chain-id', 'block-height', 'block-time', 'prev-block-hash', 'sender', 'gas-limit', 'gas-price', and 'gas-fee'.
+
+Use `chain-data` to retrieve the public metadata for a transaction. 
+This function returns an object with the following fields:
+
+- `chain-id`: The chain identifier (0-19) for the blockchain where the transaction was executed.
+- `block-height`: The height of the block that includes the transaction.
+- `block-time`: The timestamp of the block that includes the transaction.
+- `prev-block-hash`: The hash of the previous block.
+- `sender`: The sender of the transaction.
+- `gas-limit`: The gas limit for the transaction.
+- `gas-price`: The gas price for the transaction.
+- `gas-fee`: The gas fee for the transaction.
 
 ### Basic syntax
 
-To retrieve transaction public metadata using `chain-data`, use the following syntax:
+To retrieve the public metadata for a transaction using `chain-data`, use the following syntax:
+
+```pact
+(chain-data)
+```
+
+### Arguments
 
-`(chain-data)`
+You can use the `chain-data` function without arguments in code that identifies the transaction that you want to return metadata for.
 
 ### Return values
 
-The `chain-data` function returns an object with the following fields of public transaction metadata:
+The `chain-data` function returns the public metadata for a transaction as an object with the following fields
 
-- `chain-id`: The ID of the blockchain.
-- `block-height`: The height of the block.
-- `block-time`: The timestamp of the block.
-- `prev-block-hash`: The hash of the previous block.
-- `sender`: The sender of the transaction.
-- `gas-limit`: The gas limit for the transaction.
-- `gas-price`: The gas price for the transaction.
-- `gas-fee`: The gas fee for the transaction.
+| Field | Type | Description
+| ----- | ---- | -----------
+| `chain-id` | string | The chain identifier (0-19) for the blockchain where the transaction was executed.
+| `block-height` | integer | The height of the block that includes the transaction.
+| `block-time` | time | The timestamp of the block that includes the transaction.
+| `prev-block-hash` | string | The hash of the previous block.
+| `sender` | string | The sender of the transaction.
+| `gas-limit` | integer | The gas limit for the transaction.
+| `gas-price` | decimal | The gas price for the transaction.
+| `gas-fee` | decimal | The gas fee for the transaction.
 
 ### Examples
 
-The following example retrieves transaction public metadata in the Pact REPL:
+If you call the `chain-data` function in the Pact REPL without providing a transaction context in the surrounding code, the function returns the object with empty placeholder fields.
+For example:
```

Comment:
I would phrase this as just "placeholder" or "default" fields instead of "empty" which implies an empty or possibly NULL value.